protocols/types/contracts/Types.sol

```diff
@@ -24,28 +24,43 @@ library Types {
 
   bytes constant internal EIP191_HEADER = "\x19\x01";
 
+  struct Rule {
+    address senderToken;          // The address of the ERC-20 token the 'sender' is sending
+    address signerToken;          // The address of the ERC-20 token the 'signer' is sending
+    uint256 maxSenderAmount;      // The maximum amount of ERC-20 token the delegate would send
+    uint256 priceCoef;            // The whole number that will be multiplied by the 10^(-priceExp) - the price coefficient
+    uint256 priceExp;             // The exponent of the price to indicate location of the decimal priceCoef * 10^(-priceExp)
+  }
+
+  struct Intent {
+    address signerToken;          // The address of the ERC-20 token to signify the token a 'signer' would send
+    address senderToken;          // The address of the ERC-20 token to signify the token a 'sender' would send
+    uint256 amount;               // The amount to take
+    bytes32 locator;              // Locator data
+  }
+
```

Comment:
I'm thinking about taking the `Rule` and the `Intent` and using them in `Delegate` and `Indexer` so that treat those inputs as objects.

Reply:
i don't think this should exist in Types.sol

Reply:
another file, with structs agreed.

Reply:
possibly a lib.

Reply:
https://github.com/airswap/airswap-protocols/issues/168

Reply:
If this shouldn't be in Types.sol then let's not put it in there in this PR and then fix it later - better to create a structs file for these 2 right now, and deal with the `Delegate` and `Indexer` updates in the new PR.